### PR TITLE
Implement Subcommands; fix errors due to Settings <-> Plugins circular imports

### PR DESCRIPTION
**Pull request by Vendicated**
Discord implements subcommands very strangely, in that each sub command is registered as a top level command with special properties subCommandPath and rootCommand. This pr makes Vencord's CommandsAPI transform subcommands to this format (at least 1 level deep, I haven't tested nested subcommands)

Another bug is that if any plugin requires SettingsAPI, then SettingsAPI is ran before plugin init, meaning that Plugins[] will be empty and the default settings merge will fail which leads to Settings.plugins.Foo being undefined. This pr also fixes that by moving that logic inside the Proxy

### CODE REVIEW

**Comment by Vendicated** [comment]
Lol what this causes a WebContents crash

**Comment by Vendicated** [comment]
Oh cause i shadowed patches and push to the array i'm looping over so it's an infinite loop

File: src/api/Commands/index.ts [hunk 1]
```diff
@@ -88,11 +93,36 @@ function modifyOpt(opt: Option | Command) {
...
-        modifyOpt(opts[i]);
+        prepareOption(opts[i]);
+    });
+    return opt;
+}
+
+// Yes, Discord registers individual commands for each subcommand
+// TODO: This probably doesn't support nested subcommands. If that is ever needed,
+// investigate
+function registerSubCommands(cmd: Command, plugin: string) {
```
Comment (Vendicated) [inline src/api/Commands/index.ts:104]: I just realised that those won't get unregistered correctly. I don't really care right now, that can be fixed later 🚎 :trollface:

**Review by MeguminSama — APPROVED** [review]

File: src/api/Commands/index.ts (hunk 1 above)
Comment (MeguminSama) [reply src/api/Commands/index.ts:104]: 🚎

**Pull request merged**